pkg/api/types.go

```diff
@@ -2700,6 +2717,10 @@ type ServiceSpec struct {
 	// +optional
 	SessionAffinity ServiceAffinity
 
+	// ClientIPAffinityConfig contains the configurations of Client IP based session affinity.
+	// +optional
+	ClientIPAffinityConfig *ClientIPAffinityConfig
```

Comment:
Glad to receive your feedback, @thockin 

> I know it is an extra level of nest, but that is closer to how other one-of blocks work. What do you think?

I like this idea. Besides, let me make sure I have fully understood your expectations. Current API type definitions are:

```
type ServiceSpec struct {
...
  SessionAffinity ServiceAffinity
  // sessionAffinityConfig contains the configurations of session affinity.
  // +optional
  SessionAffinityConfig *SessionAffinityConfig
}

// SessionAffinityConfig represents the configurations of session affinity.
type SessionAffinityConfig struct {
	// ClientIP contains the configurations of Client IP based session affinity.
	// +optional
	ClientIP *ClientIPConfig
}

// ClientIPConfig represents the configurations of Client IP based session affinity.
type ClientIPConfig struct {
	// TimeoutSeconds specifies the seconds of ClientIP type session sticky time.
	// The value must be >0 && <=86400(for 1 day) if ServiceAffinity == "ClientIP".
	// Default value is 10800(for 3 hours).
	// Required
	TimeoutSeconds int32
}
```

Please let me know if I missed some points or if you have better ideas on naming?